server/protocols/protocols-managesieve/src/test/java/org/apache/james/managesieveserver/AuthenticateTest.java

```diff
@@ -0,0 +1,220 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+public class AuthenticateTest {
+    private ManageSieveClient client;
+    private final ManageSieveServerTestSystem testSystem;
+
+    public AuthenticateTest() throws Exception {
+        this.testSystem = new ManageSieveServerTestSystem();
+    }
+
+    @BeforeEach
+    void setUp() throws Exception {
+        this.testSystem.setUp();
+        this.client = new ManageSieveClient();
+        this.client.connect(this.testSystem.getBindedIP(), this.testSystem.getBindedPort());
+        this.client.readResponse();
+    }
+
+    @AfterEach
+    void tearDown() {
+        this.testSystem.manageSieveServer.destroy();
+    }
+
+    @Test
+    void plainLoginWithCorrectCredentialsShouldSucceed() throws IOException {
+        this.authenticatePlain();
+    }
+
+    @Test
+    void plainLoginWithWrongPasswordShouldNotSucceed() throws IOException {
+        String initialClientResponse = "\0" + ManageSieveServerTestSystem.USERNAME.asString() + "\0" + ManageSieveServerTestSystem.PASSWORD + "wrong";
+        this.client.sendCommand("AUTHENTICATE \"PLAIN\" \"" + Base64.getEncoder().encodeToString(initialClientResponse.getBytes(StandardCharsets.UTF_8)) + "\"");
+        ManageSieveClient.ServerResponse authenticationResponse = this.client.readResponse();
+        Assertions.assertThat(authenticationResponse.responseType()).isEqualTo(ManageSieveClient.ResponseType.NO);
+    }
+
+    @Test
+    void plainLoginWithNotExistingUserShouldNotSucceed() throws IOException {
+        String initialClientResponse = "\0" + ManageSieveServerTestSystem.USERNAME.asString() + "not-existing" + "\0" + "pwd";
+        this.client.sendCommand("AUTHENTICATE \"PLAIN\" \"" + Base64.getEncoder().encodeToString(initialClientResponse.getBytes(StandardCharsets.UTF_8)) + "\"");
+        ManageSieveClient.ServerResponse authenticationResponse = this.client.readResponse();
+        Assertions.assertThat(authenticationResponse.responseType()).isEqualTo(ManageSieveClient.ResponseType.NO);
+    }
+
+    @Test
+    void plainLoginWithoutPasswordShouldNotSucceed() throws IOException {
+        String initialClientResponse = "\0" + ManageSieveServerTestSystem.USERNAME.asString() + "\0";
+        this.client.sendCommand("AUTHENTICATE \"PLAIN\" \"" + Base64.getEncoder().encodeToString(initialClientResponse.getBytes(StandardCharsets.UTF_8)) + "\"");
+        ManageSieveClient.ServerResponse authenticationResponse = this.client.readResponse();
+        Assertions.assertThat(authenticationResponse.responseType()).isEqualTo(ManageSieveClient.ResponseType.NO);
+    }
+
+    // The SASL PLAIN standard (https://datatracker.ietf.org/doc/html/rfc4616) defines the following message:
+    // message = [authzid] UTF8NUL authcid UTF8NUL passwd
+    // The current code is more lenient and accepts the message without the first null byte.
+    @Disabled
+    @Test
```

Comment:
I we think it is a good thing to be more relaxed than the RFC, I see no issue:

 - Keeping the valuable comment
 - Testing the existing behaviour with a passing test.